systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -281,20 +296,255 @@ public void testCustomAndUpdatedValues() {
         }
     }
 
+    /**
+     * Test sending messages over plain transport, without auth
+     */
     @Test
     @JUnitGroup(name = "regression")
-    @KafkaFromClasspathYaml()
-    @Topic(name = TOPIC_NAME, clusterName = "my-cluster")
-    public void testSendMessages() {
+    public void testSendMessagesPlainAnonymous() throws InterruptedException {
+        String name = "send-messages-plain-anon";
         int messagesCount = 20;
-        sendMessages(kafkaPodName(CLUSTER_NAME, 1), CLUSTER_NAME, TOPIC_NAME, messagesCount);
-        String consumedMessages = consumeMessages(CLUSTER_NAME, TOPIC_NAME, 1, 30, 2);
 
-        assertThat(consumedMessages, hasJsonPath("$[*].count", hasItem(messagesCount)));
-        assertThat(consumedMessages, hasJsonPath("$[*].partitions[*].topic", hasItem(TOPIC_NAME)));
+        resources().kafkaEphemeral(CLUSTER_NAME, 3).done();
+        resources().topic(CLUSTER_NAME, TOPIC_NAME).done();
+
+        // Create ping job
+        Job job = waitForJobSuccess(pingJob(name, TOPIC_NAME, messagesCount, null, false));
 
+        // Now get the pod logs (which will be both producer and consumer logs)
+        checkPings(messagesCount, job);
     }
 
+    /**
+     * Test sending messages over tls transport using mutual tls auth
+     */
+    @Test
+    @JUnitGroup(name = "acceptance")
+    public void testSendMessagesTlsAuthenticated() {
+        String kafkaUser = "my-user";
+        String name = "send-messages-tls-auth";
+        int messagesCount = 20;
+
+        KafkaListenerAuthenticationTls auth = new KafkaListenerAuthenticationTls();
+        KafkaListenerTls listenerTls = new KafkaListenerTls();
+        listenerTls.setAuthentication(auth);
+
+        // Use a Kafka with plain listener disabled
+        resources().kafka(resources().defaultKafka(CLUSTER_NAME, 3)
+                .editSpec()
+                    .editKafka()
+                        .withNewListeners()
+                            .withTls(listenerTls)
+                            .withNewTls()
+                            .endTls()
+                        .endListeners()
```

Comment:
is there any way to avoid `end...()` methods?

Reply:
No. We're simply using the pattern that [sundrio](https://github.com/sundrio/sundrio) adopts, which is very type safe, but does mean every builder-like thing has an explicit ending method (`build()` for a top level `Builder`, `done()` for a `Doneable` and `end*()` for a nested builder as we have here).